src/Common/ZooKeeper/ZooKeeperCommon.h

```diff
@@ -25,13 +23,11 @@ struct ZooKeeperResponse : virtual Response
 {
     XID xid = 0;
 
-    UInt64 response_created_time_ns = 0;
-
     ZooKeeperResponse() = default;
     ZooKeeperResponse(const ZooKeeperResponse &) = default;
-    ~ZooKeeperResponse() override;
     virtual void readImpl(ReadBuffer &) = 0;
     virtual void writeImpl(WriteBuffer &) const = 0;
+    virtual size_t sizeImpl() const = 0;
```

Comment:
In the class `Response` we have function `virtual size_t bytesSize() const { return 0; }`

Why do we need a new `sizeImpl` function?

Reply:
It's a slightly different functionality and maybe it's not clear from the name
`sizeImpl` will return the size of serialization while `byteSize` is size of in-memory object.